### DISCUSSION TITLE
history stack is weird after redirection from action

### DISCUSSION THREAD

**Discussion by lili21** [remix-run/remix#4428] [Q&A] [answered]
let's say we got an application contains 2 pages/routes. 

One is the list page, and the other one is the create form page. Just like the below image. I can navigate to create page from list page. And after I submit, I will redirect to list page ( inside the action).

![image](https://user-images.githubusercontent.com/4456380/198176358-0fa3b6ef-2f98-485f-988f-55da39c1ddfa.png)

After the redirection, the history stack would be `['list', 'list']`, which is kinda of weird. Cause now when user click the back button, the will see the same page. 

and another unexpected behavior is, when use click back button, the loader of list page would be triggered, even it' the same page.

is it a bug?

**Comment by sergiodxa** [comment, marked as answer]
It's not expected, what should happen is the history stack is `["GET list", "GET create", "GET list"]`, the redirect after create should remove the `POST create` entry with the `GET list` entry, but if you aded `replace={true}` to the Form I think that would replace `GET create` with `GET list` too.

**Reply by lili21** [reply]
Hum. I'm using `fetcher.submit`.
